arch/arm/core/userspace.S

```diff
@@ -93,6 +93,12 @@ SECTION_FUNC(TEXT,_arm_userspace_enter)
     mrs ip, CONTROL
     orrs ip, ip, #1
     msr CONTROL, ip
+#ifndef CONFIG_CPU_CORTEX_M0
```

Comment:
In general, and if applicable: could we use "architecture" #ifdef guards instead of "cortex variants"?